src/canopy/knowledge_base/knowledge_base.py

```diff
@@ -668,17 +639,13 @@ def from_config(cls,
             )
         config['params']['index_name'] = index_name
 
-        # If the config includes an 'index_params' key, they need to be saved until
-        # the index is created, and then passed to the index creation method.
-        index_params = config['params'].pop('index_params', {})
         kb = cls._from_config(config)
-        kb._index_params = index_params
         return kb
 
-    @staticmethod
-    def _is_starter_env():
-        starter_env_suffixes = ("starter", "stage-gcp-0")
-        return os.getenv("PINECONE_ENVIRONMENT").lower().endswith(starter_env_suffixes)
+    @lru_cache(maxsize=1)
+    def _is_serverless_env(self):
+        description = self._pinecone_client.describe_index(self.index_name)
+        return "serverless" in description["spec"]
 
```

Comment:
@izellevy `KnowledgeBase` has a dedicated `from_config()` constructor, which does some stuff with `index_params`.  That part should be removed.  

(In the near future we should probably remove this method altogether, but that's a different story..)

Reply:
Yeah I removed that part.